src/Microsoft.Extensions.FileProviders.Physical/Internal/FileSystemInfoHelper.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Extensions.FileProviders.Physical.Internal
+{
+    /// <summary>
+    /// Helpful methods for FileSystem APIs.
+    /// </summary>
+    public static class FileSystemInfoHelper
```

Comment:
Can this type stay `internal`? We kind of mostly regret having so many 'pub-ternal' types, so we'd rather not add too many more...